karapace/backup/api.py

```diff
@@ -384,14 +391,17 @@ def _handle_producer_send(
         instruction.key,
         instruction.value,
     )
-    producer.send(
-        instruction.topic_name,
-        key=instruction.key,
-        value=instruction.value,
-        partition=instruction.partition_index,
-        headers=[(key.decode() if key is not None else None, value) for key, value in instruction.headers],
-        timestamp_ms=instruction.timestamp,
-    ).add_errback(_raise_backup_error)
+    try:
+        producer.send(
+            instruction.topic_name,
+            key=instruction.key,
+            value=instruction.value,
+            partition=instruction.partition_index,
+            headers=[(key.decode() if key is not None else None, value) for key, value in instruction.headers],
+            timestamp_ms=instruction.timestamp,
+        ).add_errback(_raise_backup_error)
+    except Exception as ex:
+        raise BackupDataRestorationError() from ex
```

Comment:
```suggestion
    except Exception as ex:
        raise BackupDataRestorationError() from ex
```

issue: I'm pretty adamantly against catching wild `Exception` without doing anything than a re-raise (catching a `MemoryError` and directly trying to instantiate an object is probably good way to hide the original exception). Do we really need to do that? I think the only thing we should catch here is `BackupError`, i.e. whatever is raised by `_raise_backup_error`.